n3fit/src/evolven3fit_new/evolve.py

```diff
@@ -0,0 +1,218 @@
+import pathlib
+import logging
+import sys
+
+import numpy as np
+from reportengine.compat import yaml
+
+from ekobox import genpdf, gen_info
+from ekomark import apply
+from eko import basis_rotation as br
+from eko import output
+from validphys.api import API
+
+from . import utils, eko_utils
+
+
+log = logging.getLogger(__name__)
+
+
+def evolve_fit(
+    conf_folder,
+    q_fin,
+    q_points,
+    op_card_dict,
+    t_card_dict,
+    force,
+    eko_path=None,
+    dump_eko=None,
+):
+    """
+    Evolves all the fitted replica in conf_folder/nnfit
+
+    Parameters
+    ----------
+
+        conf_folder: str or pathlib.Path
+            path to the folder containing the fit
+        q_fin: float
+            final point of the q_grid
+        q_points: int
+            number of points in the q_grid
+        op_card_dict: dict
+            user settings for the op_card
+        t_card_dict: dict
+            user settings for the t_card
+        force: bool
+            whether to force the evolution to be done again
+        eko_path: str or pathlib.Path
+            path where the eko is stored (if None the eko will be
+            recomputed)
+        dump_eko: str or pathlib.Path
+            path where the eko is dumped (if None the eko won't be
+            stored)
+    """
+    log_file = pathlib.Path(conf_folder) / "evolven3fit_new.log"
+    if log_file.exists():
+        if force:
+            log_file.unlink()
+        else:
+            raise FileExistsError(
+                f"Log file already exists: {log_file} evolven3fit_new has already been run?"
+            )
+    log_file = logging.FileHandler(log_file)
+    stdout_log = logging.StreamHandler(sys.stdout)
+    log_file.setLevel(logging.INFO)
+    stdout_log.setLevel(logging.INFO)
+    log_file.setFormatter(
+        logging.Formatter("%(asctime)s %(name)s/%(levelname)s: %(message)s")
+    )
+    stdout_log.setFormatter(
+        logging.Formatter("%(asctime)s %(name)s/%(levelname)s: %(message)s")
+    )
```

Comment:
If possible, store all values in a dictionary top level. Since you are setting the same options for the two loggers, iterate over them, instead of duplicating instructions.

Reply:
Is there a reason why you marked as solved, but doing nothing? You can also reject, but add a comment :)

Reply:
No sorry I have just misclicked it seems :)